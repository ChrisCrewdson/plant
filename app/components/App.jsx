const React = require('react');
const actions = require('../actions');
const PropTypes = require('prop-types');

class App extends React.Component {
  static contextTypes = {
    store: PropTypes.object.isRequired,
  };

  componentWillMount() {
    const { store } = this.context;
    const users = store.getState().get('users');
    if (!users || users.size === 0) {
      store.dispatch(actions.loadUsersRequest());
    }

    const locations = store.getState().get('locations');
    if (!locations || locations.size === 0) {
      store.dispatch(actions.loadLocationsRequest());
    }

    // const user = store.getState().get('user');
    // if(user && user.get('isLoggedIn', false)) {
    //   const locationIds = user.get('locationIds');
    //   if(!locationIds) {
    //     const userId = user.get('_id');
    //     store.dispatch(actions.loadLocationsRequest(userId));
    //   }
    // }
  }

  render() {
    return (
      <div className="react-root">
        {this.props.children}
      </div>
    );
  }
}

App.propTypes = {
  // eslint-disable-next-line react/forbid-prop-types
  children: PropTypes.object.isRequired,
};

module.exports = App;
