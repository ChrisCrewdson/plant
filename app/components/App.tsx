import React from 'react';
import PropTypes from 'prop-types';
import { actionFunc } from '../actions';

export default class App extends React.Component {
  // TODO: When tsc 3.7+ is in use remove the ! to see hint text on how to change this.
  context!: PlantContext;

  static propTypes = {
    // eslint-disable-next-line react/forbid-prop-types
    children: PropTypes.object.isRequired,
  };

  static contextTypes = {
    // eslint-disable-next-line react/forbid-prop-types
    store: PropTypes.object.isRequired,
  };

  // eslint-disable-next-line camelcase
  UNSAFE_componentWillMount() {
    const { store } = this.context;
    const { users = {}, locations = {} } = store.getState();

    // TODO: This will cause a problem for a non-initialized site
    // that has zero users and zero locations as these values
    // will be 0 which is falsy.
    const usersCount = Object.keys(users).length;
    const locationsCount = Object.keys(locations).length;

    if (!usersCount) {
      store.dispatch(actionFunc.loadUsersRequest());
    }

    if (!locationsCount) {
      store.dispatch(actionFunc.loadLocationsRequest());
    }
  }

  render() {
    const { children } = this.props;
    return (
      <div className="react-root">
        {children}
      </div>
    );
  }
}
