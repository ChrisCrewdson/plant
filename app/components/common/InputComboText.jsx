const PropTypes = require('prop-types');
const React = require('react');
const TextField = require('material-ui/TextField').default;

function inputComboText(props = {}) {
  const {
    changeHandler,
    disabled = false,
    error,
    fullWidth = true,
    id,
    label,
    multiLine = false,
    name: namo,
    placeholder,
    style = {},
    type = 'text',
    value,
  } = props;

  const underlineStyle = {
    display: 'none',
  };

  const styler = Object.assign({
    marginLeft: 20,
  }, style);

  return (<TextField
    disabled={disabled}
    errorText={error}
    floatingLabelText={label}
    fullWidth={fullWidth}
    hintText={placeholder}
    id={id}
    multiLine={multiLine}
    name={namo}
    onChange={changeHandler}
    style={styler}
    type={type}
    underlineStyle={underlineStyle}
    value={value}
  />);
}

inputComboText.propTypes = {
  changeHandler: PropTypes.func.isRequired,
  disabled: PropTypes.bool,
  error: PropTypes.string,
  fullWidth: PropTypes.bool,
  id: PropTypes.string.isRequired,
  label: PropTypes.string,
  multiLine: PropTypes.bool,
  name: PropTypes.string.isRequired,
  placeholder: PropTypes.string,
  // eslint-disable-next-line react/forbid-prop-types
  style: PropTypes.object,
  value: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number,
    PropTypes.bool,
  ]).isRequired,
  type: PropTypes.string,
};

inputComboText.defaultProps = {
  disabled: false,
  error: '',
  fullWidth: true,
  label: '',
  multiLine: false,
  placeholder: '',
  style: {},
  type: 'text',
};

module.exports = inputComboText;
