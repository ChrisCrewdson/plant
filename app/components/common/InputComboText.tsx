import PropTypes from 'prop-types';
import React, { ChangeEvent } from 'react';

import TextField from '@material-ui/core/TextField';
import { InputLabelProps } from '@material-ui/core/InputLabel';
import { FormHelperTextProps } from '@material-ui/core/FormHelperText';
import { InputProps } from '@material-ui/core/Input';

type InputComboTextPropsType = 'text' | 'number';

export interface InputComboTextProps {
  changeHandler: (name: string, newValue: string) => void;
  disabled?: boolean;
  error: React.ReactNode;
  fullWidth?: boolean;
  id: string;
  label: React.ReactNode;
  multiLine?: boolean;
  name: string;
  placeholder?: string;
  style?: React.CSSProperties;
  type?: InputComboTextPropsType;
  value: string | number;
}


/**
 * A very light wrapper around the MaterialUI TextField component
 */
export default function inputComboText(props: InputComboTextProps): JSX.Element {
  const {
    changeHandler,
    disabled = false,
    error: errorText,
    fullWidth = true,
    id,
    label,
    multiLine = false,
    name: namo,
    placeholder,
    style = {},
    type = 'text',
    value,
  } = props;

  // const underlineStyle = {
  //   display: 'none',
  // };

  const onChange = (e: ChangeEvent<HTMLInputElement | HTMLTextAreaElement>): void => {
    changeHandler(e.target.name, e.target.value);
  };

  const fontSize = '1.5em';
  const styler = {
    marginLeft: 20, marginRight: 20, marginBottom: 20, ...style,
  };
  const error = !!errorText;

  // These "sub" styles are for the sub-components that are wrapped
  // inside the TextField component and will be pushed down to their
  // respective components.
  const subStyle = {
    style: { fontSize },
  };
  const subInputLabelProps: InputLabelProps = subStyle;
  const subFormHelperTextProps: FormHelperTextProps = subStyle;
  const subInputProps: InputProps = subStyle;

  return (
    <TextField
      disabled={disabled}
      error={error}
      FormHelperTextProps={subFormHelperTextProps}
      fullWidth={fullWidth}
      helperText={errorText}
      id={id}
      InputLabelProps={subInputLabelProps}
      InputProps={subInputProps}
      label={label}
      multiline={multiLine}
      name={namo}
      onChange={onChange}
      placeholder={placeholder}
      style={styler}
      type={type}
      value={value}
      variant="standard"
    />
  );
}

inputComboText.propTypes = {
  changeHandler: PropTypes.func.isRequired,
  disabled: PropTypes.bool,
  error: PropTypes.string,
  fullWidth: PropTypes.bool,
  id: PropTypes.string.isRequired,
  label: PropTypes.string,
  multiLine: PropTypes.bool,
  name: PropTypes.string.isRequired,
  placeholder: PropTypes.string,
  // eslint-disable-next-line react/forbid-prop-types
  style: PropTypes.object,
  value: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number,
    PropTypes.bool,
  ]).isRequired,
  type: PropTypes.string,
};

inputComboText.defaultProps = {
  disabled: false,
  error: '',
  fullWidth: true,
  label: '',
  multiLine: false,
  placeholder: '',
  style: {},
  type: 'text',
};
