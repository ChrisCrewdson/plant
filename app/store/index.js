const { createStore, applyMiddleware } = require('redux');
const reducers = require('../reducers'); // combineReducers already called on reducers in her)e
const api = require('../middleware/api');
const { setupSubscribe: userSubscribe } = require('./user');
const Immutable = require('immutable');

let middleware = [api];

if (process.env.NODE_ENV !== 'production') {
  const logger = require('../middleware/logger');
  middleware = [logger, api];
}

// Add the api to the pipeline/chain
const createStoreWithMiddleware = applyMiddleware(...middleware)(createStore);

const store = createStoreWithMiddleware(reducers, Immutable.fromJS(window.__INITIAL_STATE__ || {}));

userSubscribe(store);

module.exports = store;
