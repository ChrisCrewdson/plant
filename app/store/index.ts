export {}; // To get around: Cannot redeclare block-scoped variable .ts(2451)

const { createStore, applyMiddleware } = require('redux');

const seamless = require('seamless-immutable').static;
const reducers = require('../reducers'); // combineReducers already called on reducers in her)e
const api = require('../middleware/api');
const { setupSubscribe: userSubscribe } = require('./user');

const middleware = [api];

if (process.env.NODE_ENV !== 'production') {
  // eslint-disable-next-line global-require
  const logger = require('../middleware/logger');
  middleware.unshift(logger);
}

// Add the api to the pipeline/chain
const createStoreWithMiddleware = applyMiddleware(...middleware)(createStore);

// @ts-ignore - __INITIAL_STATE__ is added on the server
const initialState = seamless.from(window.__INITIAL_STATE__ || {});
const store = createStoreWithMiddleware(reducers, initialState);

userSubscribe(store);

module.exports = store;
