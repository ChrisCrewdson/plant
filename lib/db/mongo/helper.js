const _ = require('lodash');

module.exports = class Helper {
  static removeEmpty(doc) {
    if (!doc) {
      return doc;
    }
    if (_.isArray(doc)) {
      return doc.map(item => Helper.removeEmpty(item));
    }
    return _.pickBy(doc, item => item !== '');
  }

  /**
   * Helper function to convert _id from MongoId to string. Used in reads
   * @param {object} obj - Object that might have an _id
   * @returns {object} - the same object with a converted _id
   */
  static convertIdToString(obj) {
    if (_.isArray(obj)) {
      return obj.map(Helper.convertIdToString);
    }
    if (obj && obj._id) {
      // eslint-disable-next-line no-param-reassign
      obj._id = obj._id.toString();
    }
    return obj;
  }
};
