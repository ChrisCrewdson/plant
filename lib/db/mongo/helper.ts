import _ from 'lodash';

export class Helper {
  /**
   * Removes empty string values from properties
   * @param doc - an object with props that might have empty string values
   */
  static removeEmpty(doc: object | Array<any>): object | Array<any> {
    if (!doc) {
      return doc;
    }
    if (_.isArray(doc)) {
      return (doc as Array<any>).map((item) => Helper.removeEmpty(item));
    }
    return _.pickBy(doc, (item: any) => item !== '');
  }

  /**
   * Helper function to convert _id from MongoId to string. Used in reads
   * @param obj - Object that might have an _id
   * @returns - the same object with a converted _id
   */
  static convertIdToString(obj: any): object {
    if (_.isArray(obj)) {
      return (obj as Array<any>).map(Helper.convertIdToString);
    }
    if (obj && obj._id) {
      // eslint-disable-next-line no-param-reassign
      obj._id = obj._id.toString();
    }
    return obj;
  }
}
