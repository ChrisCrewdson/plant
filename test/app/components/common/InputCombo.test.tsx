import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import { Dispatch } from 'redux';

import { MuiThemeProvider } from '@material-ui/core/styles';

import InputComboText from '../../../../app/components/common/InputComboText';
import store from '../../../../app/store';
import App from '../../../../app/components/App';
import { PlantAction } from '../../../../lib/types/redux-payloads';
import { theme } from '../../../helper';

describe('InputComboText and SelectCombo', () => {
  const storeDispatch = store.dispatch;

  beforeAll(() => {
    store.dispatch = (() => {}) as Dispatch<PlantAction<any>>;
  });

  afterAll(() => {
    store.dispatch = storeDispatch;
  });

  test('should render Text Input', () => {
    const changeHandler = () => {};
    const component = renderer.create(
      <MuiThemeProvider theme={theme}>
        <Provider store={store}>
          <App>
            <MemoryRouter>
              <InputComboText
                changeHandler={changeHandler}
                id="test-id"
                name="test-name"
                placeholder="placeholder text"
                value="test-value"
              />
            </MemoryRouter>
          </App>
        </Provider>
      </MuiThemeProvider>);
    const badgeTotals = component.toJSON();
    expect(badgeTotals).toMatchSnapshot();
  });
});
