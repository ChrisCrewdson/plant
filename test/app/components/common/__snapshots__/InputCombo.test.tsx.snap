// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputComboText and SelectCombo should render Text Input 1`] = `
<div
  className="react-root"
>
  <div
    className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
    style={
      Object {
        "marginBottom": 20,
        "marginLeft": 20,
        "marginRight": 20,
      }
    }
  >
    
    <div
      className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
      onClick={[Function]}
      style={
        Object {
          "fontSize": "1.5em",
        }
      }
    >
      <input
        aria-invalid={false}
        autoFocus={false}
        className="MuiInputBase-input MuiInput-input"
        disabled={false}
        id="test-id"
        name="test-name"
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        placeholder="placeholder text"
        required={false}
        type="text"
        value="test-value"
      />
    </div>
    
  </div>
</div>
`;
